app/models/good_job/execution.rb

```diff
@@ -345,9 +345,9 @@ def perform
             end
             handled_error ||= current_thread.error_on_retry || current_thread.error_on_discard
 
-            ExecutionResult.new(value: value, handled_error: handled_error, retried: current_thread.error_on_retry.present?)
+            instrument_payload[:execution_result] = ExecutionResult.new(value: value, handled_error: handled_error, retried: current_thread.error_on_retry.present?)
```

Comment:
As part of the public interface, I want to be explicit

```suggestion
            instrument_payload.merge!(
              value: value, 
              handled_error: handled_error, 
              retried: current_thread.error_on_retry.present?
            )
```